smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/ExtractTelemetryLoggerConfig.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.smithy.swift.codegen.integration
+
+import software.amazon.smithy.swift.codegen.ClientRuntimeTypes
+import software.amazon.smithy.swift.codegen.SwiftWriter
+import software.amazon.smithy.swift.codegen.core.GenerationContext
+import software.amazon.smithy.swift.codegen.integration.HttpProtocolServiceClient.ConfigClassVariablesCustomization
+import software.amazon.smithy.swift.codegen.integration.HttpProtocolServiceClient.ConfigInitializerCustomization
+import software.amazon.smithy.utils.CodeInterceptor
+import software.amazon.smithy.utils.CodeSection
+
+class ExtractTelemetryLoggerConfig : SwiftIntegration {
+    override fun interceptors(codegenContext: GenerationContext): List<out CodeInterceptor<out CodeSection?, SwiftWriter>?> {
+        return listOf(InternalLoggerConfigVariable(), LoggerConfigInitializer())
+    }
+
+    private class InternalLoggerConfigVariable : CodeInterceptor.Appender<ConfigClassVariablesCustomization, SwiftWriter> {
+        override fun sectionType(): Class<ConfigClassVariablesCustomization> {
+            return ConfigClassVariablesCustomization::class.java
+        }
+
+        override fun append(writer: SwiftWriter, section: ConfigClassVariablesCustomization) {
+            writer.write("internal let logger: \$N", ClientRuntimeTypes.Core.Logger)
```

Comment:
Q: How does writer know to write this at the given section when it's not used in the method call? Same for the `ConfigInitializerCustomization` below.